### PR TITLE
Fixes 3357: Add UUID/URL filter for list repositories

### PR DESCRIPTION
**Pull request by Andrewgdewar**
## Summary

- Adds 1 new variable  (UUID)  to the FilterData struct for repoconfigs.
- Updates 1 variable (URL) to now accept a comma separated string list.

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-3357

**Comment by swadeley** [comment]
Hi

Looks good (url and uuid)

`In [5]: app.content_sources.rest_client.repositories_api.list_repositories(uuid='87c6c5d6-fb59-4b6f-ba3e-81446c5efbe8,7999300c-6063-43d6-a685-054c9b700d87')
2024-01-16 08:49:10.965 [    INFO] [iqe.base.rest_client] REST: METHOD=GET, <snip>, params=[('uuid', '87c6c5d6-fb59-4b6f-ba3e-81446c5efbe8,7999300c-6063-43d6-a685-054c9b700d87')]
Out[5]: 
{'data': [{'account_id': '0369233',
           'content_type': 'rpm',
           'distribution_arch': 'any',
           'distribution_versions': ['any'],
           'failed_introspections_count': 0,
           'gpg_key': '',
           'last_introspection_error': '',
           'last_introspection_time': '2024-01-16T08:33:35Z',
           'last_success_introspection_time': '2024-01-16T08:33:35Z',
           'last_update_introspection_time': '2024-01-16T08:00:22Z',
           'metadata_verification': False,
           'module_hotfixes': False,
           'name': 'comps-env-test',
           'org_id': '3340851',
           'origin': 'external',
           'package_count': 10003,
           'snapshot': False,
           'status': 'Valid',
           'url': 'https://dl.fedoraproject.org/pub/epel/8/Everything/x86_64/',
           'uuid': '87c6c5d6-fb59-4b6f-ba3e-81446c5efbe8'},
          {'account_id': '0369233',
           'content_type': 'rpm',
           'distribution_arch': 'any',
           'distribution_versions': ['any'],
           'failed_introspections_count': 0,
           'gpg_key': 'https://jlsherrill.fedorapeople.org/fake-repos/signed/GPG-KEY.gpg',
           'last_introspection_error': '',
           'last_introspection_time': '2024-01-16T08:48:28Z',
           'last_success_introspection_time': '2024-01-16T08:48:28Z',
           'last_update_introspection_time': '2024-01-16T08:48:28Z',
           'metadata_verification': False,
           'module_hotfixes': False,
           'name': 'gpgtest-snapshot-false',
           'org_id': '3340851',
           'origin': 'external',
           'package_count': 33,
           'snapshot': False,
           'status': 'Valid',
           'url': 'http://jlsherrill.fedorapeople.org/fake-repos/signed/',
           'uuid': '7999300c-6063-43d6-a685-054c9b700d87'}],
 'links': {'first': '/api/content-sources/v1/repositories/?limit=100&offset=0&uuid=87c6c5d6-fb59-4b6f-ba3e-81446c5efbe8,7999300c-6063-43d6-a685-054c9b700d87',
           'last': '/api/content-sources/v1/repositories/?limit=100&offset=0&uuid=87c6c5d6-fb59-4b6f-ba3e-81446c5efbe8,7999300c-6063-43d6-a685-054c9b700d87'},
 'meta': {'count': 2, 'limit': 100, 'offset': 0}}

In [6]: app.content_sources.rest_client.repositories_api.list_repositories(url='https://dl.fedoraproject.org/pub/epel/8/Everything/x86_64/,http://jlsherrill.fedorapeople.org/fake-repos/signed/')
2024-01-16 09:11:33.043 [    INFO] [iqe.base.rest_client] REST: METHOD=GET, <snip>, params=[('url', 'https://dl.fedoraproject.org/pub/epel/8/Everything/x86_64/,http://jlsherrill.fedorapeople.org/fake-repos/signed/')]
Out[6]: 
{'data': [{'account_id': '0369233',
           'content_type': 'rpm',
           'distribution_arch': 'any',
           'distribution_versions': ['any'],
           'failed_introspections_count': 0,
           'gpg_key': '',
           'last_introspection_error': '',
           'last_introspection_time': '2024-01-16T08:33:35Z',
           'last_success_introspection_time': '2024-01-16T08:33:35Z',
           'last_update_introspection_time': '2024-01-16T08:00:22Z',
           'metadata_verification': False,
           'module_hotfixes': False,
           'name': 'comps-env-test',
           'org_id': '3340851',
           'origin': 'external',
           'package_count': 10003,
           'snapshot': False,
           'status': 'Valid',
           'url': 'https://dl.fedoraproject.org/pub/epel/8/Everything/x86_64/',
           'uuid': '87c6c5d6-fb59-4b6f-ba3e-81446c5efbe8'},
          {'account_id': '0369233',
           'content_type': 'rpm',
           'distribution_arch': 'any',
           'distribution_versions': ['any'],
           'failed_introspections_count': 0,
           'gpg_key': 'https://jlsherrill.fedorapeople.org/fake-repos/signed/GPG-KEY.gpg',
           'last_introspection_error': '',
           'last_introspection_time': '2024-01-16T08:48:28Z',
           'last_success_introspection_time': '2024-01-16T08:48:28Z',
           'last_update_introspection_time': '2024-01-16T08:48:28Z',
           'metadata_verification': False,
           'module_hotfixes': False,
           'name': 'gpgtest-snapshot-false',
           'org_id': '3340851',
           'origin': 'external',
           'package_count': 33,
           'snapshot': False,
           'status': 'Valid',
           'url': 'http://jlsherrill.fedorapeople.org/fake-repos/signed/',
           'uuid': '7999300c-6063-43d6-a685-054c9b700d87'}],
 'links': {'first': '/api/content-sources/v1/repositories/?limit=100&offset=0&url=https://dl.fedoraproject.org/pub/epel/8/Everything/x86_64/,http://jlsherrill.fedorapeople.org/fake-repos/signed/',
           'last': '/api/content-sources/v1/repositories/?limit=100&offset=0&url=https://dl.fedoraproject.org/pub/epel/8/Everything/x86_64/,http://jlsherrill.fedorapeople.org/fake-repos/signed/'},
 'meta': {'count': 2, 'limit': 100, 'offset': 0}}
`

**Comment by swadeley** [comment]
Hi @jlsherrill 

I see pulp start up failed, so pr checks failed.

File: api/openapi.json [hunk 1]
```diff
@@ -1434,13 +1434,21 @@
...
-                        "description": "Filter repositories by URL.",
+                        "description": "A comma separated list of URLs to control api response.",
                         "in": "query",
                         "name": "url",
                         "schema": {
                             "type": "string"
                         }
                     },
+                    {
+                        "description": "A comma separated list of  uuids to control api response.",
```
Comment (swadeley) [inline api/openapi.json:1445]:
```suggestion
                        "description": "A comma separated list of uuids to control api response.",
```

File: api/docs.go [hunk 2]
```diff
@@ -377,10 +377,16 @@ const docTemplate = `{
...
                     {
                         "type": "string",
-                        "description": "Filter repositories by URL.",
+                        "description": "A comma separated list of URLs to control api response.",
                         "name": "url",
                         "in": "query"
                     },
+                    {
+                        "type": "string",
+                        "description": "A comma separated list of  uuids to control api response.",
```
Comment (swadeley) [inline api/docs.go:386]:
```suggestion
                        "description": "A comma separated list of uuids to control api response.",
```

File: pkg/handler/repositories.go [hunk 3]
```diff
@@ -92,7 +92,8 @@ func getAccountIdOrgId(c echo.Context) (string, string) {
 // @Param		 available_for_arch query string false "Filter repositories by architecture. For example, `x86_64` returns repositories with the version `x86_64` or where architecture is not set."
 // @Param		 search query string false "Term to filter and retrieve items that match the specified search criteria. Search term can include name or URL."
 // @Param		 name query string false "Filter repositories by name."
-// @Param		 url query string false "Filter repositories by URL."
+// @Param		 url query string false "A comma separated list of URLs to control api response."
+// @Param		 uuid query string false "A comma separated list of  uuids to control api response."
```
Comment (swadeley) [inline pkg/handler/repositories.go:96]:
```suggestion
// @Param		 uuid query string false "A comma separated list of uuids to control api response."
```

**Comment by swadeley** [comment]
Hi, apidoc for QE updated with latest changes. Thank you.

**Pull request merged**